source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. tip::
+
+   While you do not get the warning about finding the nv file when using the ``--nv`` flag, you also have to specify which GPU to use, otherwise none are allocated to you! This can be done with the ``-G`` flag, as can be seen in the example shell script. 
+
+Now you are ready to build on top of a base container and run your code on a GPU!
+
+.. _accounting-gpu:
+
+Accounting of GPU usage
+=======================
+
+Currently the usage of GPU nodes is accounted for in GPU hours. This means that even though multiple cores are used simultaneously, one hour of use of a GPU node is billed as 1 GPU-hour. Unlike CPUs where one hour of multi-core usage is billed as multiple CPU hours, depending on the number of cores.
+
+.. _singularity-building:
+
+================================
+Building and running a singularity container
+================================
+
+In this section we show how to build a singularity container use it to run code in its environment. There is extensive documentation from singularity itself `here <https://docs.sylabs.io/guides/latest/user-guide/index.html>`_. 
+
+The steps in this section are done on GPU nodes, to ensure availability of the drivers, which may be needed in some compilation steps.
+
+Building directly from dockerhub
+================================
+
+There are multiple ways to build a container. To build directly from docker hub, for example the latest version of tensorflow, one can invoke:
+
+.. code-block:: bash
+   
+   singularity build --nv tf_latest.sif docker://tensorflow/tensorflow:latest
+
+and the image ``tf_latest.sif`` from `dockerhub <https://hub.docker.com>`_ will be built, containing the contents of the latest ``tensorflow`` image from the makers of tensorflow. You can also get an image from a different source, such as the Nvidia container repository:
+
+.. code-block:: bash
+
+   singularity build --nv nvidia-tf.sif docker://nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+Such an image will contain all the necessary drivers and compile flags to run on Nvidia GPUs, which is preferable on :abbr:`Spider (Symbiotic Platform(s) for Interoperable Data Extraction and Redistribution)`.
+
+To directly run the container without writing to disk invoke:
+
+.. code-block:: bash
+
+   singularity run --nv docker://nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+In the examples below, the base images are taken from the internet and expanded upon using *definition* files, to build custom singularity containers.
+
+Running CUDA code 
+=================
+
+
+Here, we show the method of using a definitions file, as opposed to above, where directly building from a repository is shown. The contents of the definitions file are shown before these contents are explained. Start by making the file called ``cuda_example.def`` and add all the steps we want to take to make a container:
```

Comment:
definition or definitions?

Perhaps earlier when I complained about missing a cuda example you could simply link this example there?